WooCommerce/src/main/kotlin/com/woocommerce/android/SiteObserver.kt

```diff
@@ -25,6 +27,14 @@ class SiteObserver @Inject constructor(
             .collect { site ->
                 WooLog.d(WooLog.T.UTILS, "Fetch plugins for site ${site.name}")
                 wooCommerceStore.fetchSitePlugins(site)
+
+                // Makes sure the store ID is fetched for the site.
+                environmentRepository.fetchOrGetStoreID(site)
+                    .takeIf { result -> result.isError.not() }
+                    ?.let { result ->
+                        val storeID = result.model ?: return@let
+                        WooLog.d(WooLog.T.UTILS, "Fetched StoreID $storeID for site ${site.name}")
+                    }
```

Comment:
np: Since you're using a `?.let` call to unwrap the `takeIf` response, you can directly add to the call stack another unwrapping with the `model` field with the `let` and simplify the verification effort:

```suggestion
environmentRepository.fetchOrGetStoreID(site)
                    .takeIf { result -> result.isError.not() }
                    ?.model?.let { storeID ->
                        WooLog.d(WooLog.T.UTILS, "Fetched StoreID $storeID for site ${site.name}")
                    }
```

Reply:
Nice tip!